### PR TITLE
Enable bulk delete modal for Languages, Taxes, CMS Pages, CMS Cats

### PR DESCRIPTION
**Pull request by matks**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | Enable bulk delete modal for Languages, Taxes, CMS Pages, CMS Pages Categories
| Type?         | improvement
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes some items from https://github.com/PrestaShop/PrestaShop/issues/14462
| How to test?  | See ticket description, this PR solves it for pages Languages, Taxes, CMS Pages, CMS Pages Categories

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/17287)

### CODE REVIEW

File: src/Core/Grid/Definition/Factory/CurrencyGridDefinitionFactory.php [hunk 1]
```diff
@@ -47,6 +47,8 @@
  */
 final class CurrencyGridDefinitionFactory extends AbstractGridDefinitionFactory
 {
+    use BulkDeleteActionTrait;
```
Comment (Progi1984) [inline src/Core/Grid/Definition/Factory/CurrencyGridDefinitionFactory.php]: @matks Why adding this trait if you not use it ?

Comment (matks) [reply src/Core/Grid/Definition/Factory/CurrencyGridDefinitionFactory.php]: I might have done some overthinking 😅 my idea was "what if a module developer WANTS to extend the behavior of the Currency Grid by adding bulk delete"

Comment (matks) [reply src/Core/Grid/Definition/Factory/CurrencyGridDefinitionFactory.php]: Removed

**Comment by matks** [comment]
@SimonGrn @boubkerbribri this will remove more `window.open()` and replace them by modals, as asked in issue #14462. This has already broken some nightly tests before and is likely to break more.

**Comment by SimonGrn** [comment]
> @SimonGrn @boubkerbribri this will remove more `window.open()` and replace them by modals, as asked in issue #14462. This has already broken some nightly tests before and is likely to break more.

Thanks for the heads-up !

**Review by sowbiba — APPROVED** [review]
:+1:

**Review by atomiix — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Pull request merged**